### PR TITLE
blk/kernel: expose IORING_SETUP_{IOPOLL,SQPOLL} as options

### PR DESCRIPTION
**Pull request by lnsyyj**
Signed-off-by: JiangYu <lnsyyj@hotmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @rouming

**Comment by rouming** [comment]
Looks good to me.

File: src/blk/kernel/KernelDevice.cc [hunk 1]
```diff
@@ -67,10 +67,12 @@ KernelDevice::KernelDevice(CephContext* cct, aio_callback_t cb, void *cbpriv, ai
   fd_buffereds.resize(WRITE_LIFE_MAX, -1);
 
   bool use_ioring = cct->_conf.get_val<bool>("bdev_ioring");
+  bool use_ioring_hipri = cct->_conf.get_val<bool>("bdev_ioring_hipri");
+  bool use_ioring_sqthread_poll = cct->_conf.get_val<bool>("bdev_ioring_sqthread_poll");
```
Comment (tchaikov) [inline src/blk/kernel/KernelDevice.cc:71]: could you move these two lines into the `if (use_ioring && ioring_queue_t::supported()) {` block?

Comment (lnsyyj) [reply src/blk/kernel/KernelDevice.cc:71]: Ok, no problem.

**Review by tchaikov — APPROVED** [review]

**Comment by lnsyyj** [comment]
Hi @tchaikov ,
Looks like the performance test failed? But I haven't adjusted the iouring interface.
I want to study liburing and modify it later.

**Comment by tchaikov** [comment]
jenkins test classic perf

**Comment by lnsyyj** [comment]
> jenkins test classic perf

I think this PR can not test the performance, not involve the IO interface, just expose the configuration.

**Comment by tchaikov** [comment]
> > jenkins test classic perf
> 
> I think this PR can not test the performance, not involve the IO interface, just expose the configuration.

agreed. i just wanted to verify the perf test works.

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2020-11-26_05:04:35-rados-wip-kefu-testing-2020-11-26-1029-distro-basic-smithi/

**Pull request merged**